libs/component-library/src/lib/radio-button/RadioButton.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  Animated,
+  View,
+  StyleSheet,
+  TouchableOpacity,
+  TouchableOpacityProps,
+} from 'react-native';
+import { FxTheme } from '../theme/theme';
+
+import type { $RemoveChildren } from './../types';
+
+import { useRadioButtonContext } from './RadioButtonGroup';
+import { handlePress, isChecked } from './utils';
+
+const TouchableBox = createBox<FxTheme, TouchableOpacityProps>(
+  TouchableOpacity
+);
+
+type Props = $RemoveChildren<typeof TouchableBox> & {
+  /**
+   * Value of the radio button
+   */
+  value: string;
+  /**
+   * Status of radio button.
+   */
+  status?: 'checked' | 'unchecked';
+  /**
+   * Whether radio is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Function to execute on press.
+   */
+  onPress?: () => void;
+  /**
+   * Custom color for radio.
+   */
+  checkedColor?: string;
+  /**
+   * Custom color for unchecked radio.
+   */
+  uncheckedColor?: string;
+  /**
+   * Custom color for radio.
+   */
+  checkedDisabledColor?: string;
+  /**
+   * Custom color for unchecked disabled radio.
+   */
+  uncheckedDisabledColor?: string;
+  /**
+   * Custom color for unchecked disabled radio.
+   */
+  uncheckedDisabledBackgroundColor?: string;
+};
+
+const BORDER_WIDTH = 1;
+const BORDER_WIDTH_CHECKED = 6;
+
+/**
+ * ## Usage
+ * ```js
+ * import * as React from 'react';
+ * import { View } from 'react-native';
+ * import { RadioButton } from '@functionland/component-library';
+ *
+ * const MyComponent = () => {
+ *   const [checked, setChecked] = React.useState('first');
+ *
+ *   return (
+ *     <View>
+ *       <RadioButton
+ *         value="first"
+ *         status={ checked === 'first' ? 'checked' : 'unchecked' }
+ *         onPress={() => setChecked('first')}
+ *       />
+ *       <RadioButton
+ *         value="second"
+ *         status={ checked === 'second' ? 'checked' : 'unchecked' }
+ *         onPress={() => setChecked('second')}
+ *       />
+ *     </View>
+ *   );
+ * };
+ *
+ * export default MyComponent;
+ * ```
+ */
+
+const RadioButton = ({ disabled, onPress, value, status, ...rest }: Props) => {
+  const { colors } = useTheme<FxTheme>();
+  const { value: contextValue, onValueChange } = useRadioButtonContext();
+  const checked =
+    isChecked({
+      contextValue,
+      status,
+      value,
+    }) === 'checked';
+  const { current: borderAnim } = React.useRef<Animated.Value>(
+    new Animated.Value(checked ? BORDER_WIDTH_CHECKED : BORDER_WIDTH)
+  );
+
+  const { current: radioAnim } = React.useRef<Animated.Value>(
+    new Animated.Value(1)
+  );
+
+  const isFirstRendering = React.useRef<boolean>(true);
+
+  React.useEffect(() => {
+    // Do not run animation on very first rendering
+    if (isFirstRendering.current) {
+      isFirstRendering.current = false;
+      return;
+    }
+
+    if (checked) {
+      radioAnim.setValue(2.5);
+
+      Animated.timing(radioAnim, {
+        toValue: 1,
+        duration: 250,
+        useNativeDriver: true,
+      }).start();
```

Comment:
Normally we've been using `Reanimated` where we can. Not worth rewriting with this already done and working!

Reply:
Okay, I'll use it in the future.